tao_compiler/mlir/disc/transforms/fusion_utils.cc

```diff
@@ -1863,6 +1894,56 @@ bool StitchCPUAnalysis::doBcastOpTileAnalysis(
   return true;
 }
 
+SmallVector<int> getNonSizeOneDims(MemRefType ty) {
+  SmallVector<int> dims;
+  for (int i = 0; i < ty.getRank(); ++i)
+    if (ty.getShape()[i] != 1) dims.push_back(i);
+  return dims;
+}
+
+bool StitchCPUAnalysis::doReshapeOpTileAnalysis(
+    DenseMap<Value, TileInfo>& tilePlan, Operation* op, bool& changed) {
+  Value inValue = op->getOperand(0);
+  Value outValue = cast<lmhlo::LmhloOp>(op).getResultBuffer();
+  // target_shape should not do parallel computation.
+  if (isa<lmhlo::DynamicReshapeOp>(op)) {
+    auto& targetShape = tilePlan[op->getOperand(1)];
+    TileInfo mergedInfo = targetShape;
+    if (!mergedInfo.merge(0)) return false;
+    changed |= targetShape.updateIfNotEqual(mergedInfo);
+  }
+
+  auto propagateTileDims = [&](Value in, TileInfo& inInfo, Value out,
```

Comment:
nit: propagatableTileDims

Reply:
using verb + noun is more common style for function name? Furthermore, this function actually also update the passed inputs.